job_executor/worker/steps/dataset_pseudonymizer.py

```diff
@@ -20,192 +26,136 @@ def _get_unit_types(
-                        stop_date,
-                    ]
-                )
-                + "\n"
-            )
-    target_file.close()
-    return output_csv_path
 
+    return [
+        identifier_to_pseudonym[identifier] for identifier in identifiers_list
+    ]
 
-def _pseudonymize_measure_only(
-    input_csv_path: Path, unit_id_type: str, job_id: str
-) -> Path:
-    unique_measure_values = set()
-    with open(input_csv_path, newline="", encoding="utf-8") as csv_file:
-        for line in csv_file:
-            value = line.strip().split(";")[2]
-            unique_measure_values.add(value)
-    value_to_pseudonym = pseudonym_service.pseudonymize(
-        list(unique_measure_values), unit_id_type, job_id
-    )
-    output_csv_path = (
-        input_csv_path.parent / f"{input_csv_path.stem}_pseudonymized.csv"
-    )
-    target_file = open(output_csv_path, "w", newline="", encoding="utf-8")
-    with open(input_csv_path, newline="", encoding="utf-8") as csv_file:
-        for line in csv_file:
-            row = line.strip().split(";")
-            line_number: int = row[0]
-            unit_id: str = row[1]
-            value: str = row[2]
-            start_date: str = row[3]
-            stop_date: str = row[4]
-            target_file.write(
-                ";".join(
-                    [
-                        str(line_number),
-                        unit_id,
-                        str(value_to_pseudonym[value]),
-                        start_date,
-                        stop_date,
-                    ]
-                )
-                + "\n"
-            )
-    target_file.close()
-    return output_csv_path
+
+def _get_columns_excluding(
+    dataset: dataset.FileSystemDataset, excluded_columns: List[str]
+) -> List[str]:
+    """
+    Get a list of column names from the dataset excluding the specified columns.
+    """
+    return [
+        name for name in dataset.schema.names if name not in excluded_columns
+    ]
 
 
-def _pseudonymize_identifier_and_measure(
-    input_csv_path: Path,
-    identifier_unit_id_type: str,
-    measure_unit_id_type: str,
+def _pseudonymize(
+    input_parquet_path: Path,
+    identifier_unit_id_type: Optional[UnitType],
+    measure_unit_id_type: Optional[UnitType],
     job_id: str,
 ) -> Path:
-    unique_idents = set()
-    unique_measure_values = set()
-    with open(input_csv_path, newline="", encoding="utf-8") as csv_file:
-        for line in csv_file:
-            row = line.strip().split(";")
-            unit_id = row[1]
-            value = row[2]
-            unique_idents.add(unit_id)
-            unique_measure_values.add(value)
-    identifier_to_pseudonym = pseudonym_service.pseudonymize(
-        list(unique_idents), identifier_unit_id_type, job_id
-    )
-    value_to_pseudonym = pseudonym_service.pseudonymize(
-        list(unique_measure_values), measure_unit_id_type, job_id
+    input_dataset = dataset.dataset(input_parquet_path)
```

Comment:
Ref: our discussion earlier

Could something in this direction work?

```py
input_dataset = dataset.dataset(input_parquet_path)

unit_id_pseudonyms = _pseudonymize_column(
	input_dataset, "unit_id", identifier_unit_id_type, job_id
)
value_pseudonyms = _pseudonymize_column(
	input_dataset, "value", measure_unit_id_type, job_id
)

column_names = input_dataset.schema.names

columns = []
for column_name in original_column_names
	if column_name == "unit_id" and unit_id_pseudonyms:
		columns.append(unit_id_pseudonyms)
		continue
	if column_name == "value" and value_pseudonyms:
		columns.append(value_pseudonyms)
		continue
	columns.append(
		input_dataset.to_table(
			columns=[column_name]
		)[column_name]

pseudonymized_table = pyarrow.Table.from_arrays(
    columns, column_names
)
```

This way we never mess with the column names at all which might be easier to reason with 🤔

Reply:
@DanielElisenberg This one is nice 👍

Reply:
This looks alot cleaner and easier to read!